### PR TITLE
feat(cli): support i18n overrides for navigation labels via YAML overlays

### PR DESCRIPTION
**Pull request by dsinghvi**
## Description

Adds support for translating navigation labels (tabs, products, versions, sections, pages, announcements) via YAML overlay files in `translations/<lang>/fern/`. The overlay files use the exact same format that `fern write-translation` already generates, so users can simply translate the values in place.

This closes the gap where `translations:` config only loaded per-language `.mdx` page content but had no mechanism for translating navigation chrome fields (tab names, product titles, version labels, section headings, etc.).

## How it works

1. Users place translated YAML overlay files at `translations/<lang>/fern/docs.yml` and referenced nav YAML files (e.g., `nav.yml`)
2. During docs publishing, the CLI loads these overlay files and extracts translatable fields into a `TranslationNavigationOverlay` structure
3. When building per-locale `DocsDefinition`, the overlay is applied to the nav tree by walking it and matching products (by slug), versions (by slug), tabs (by ID/slug), and sections/pages (by slug or position)
4. Translated announcement messages are also applied to the per-locale config

### Example

Source `docs.yml`:
```yaml
products:
  - display-name: Home
    path: ./nav.yml
    slug: home
```

Translation overlay at `translations/ja/fern/docs.yml`:
```yaml
products:
  - display-name: ホーム
    path: ./nav.yml
    slug: home
```

Translation overlay at `translations/ja/fern/nav.yml`:
```yaml
tabs:
  docs:
    display-name: ドキュメント
navigation:
  - tab: docs
    layout:
      - page: はじめに
        path: ./pages/getting-started.mdx
        slug: getting-started
      - section: APIガイド
        contents:
          - page: 概要
            path: ./pages/api/overview.mdx
            slug: overview
        slug: api-guide
```

## Changes Made

- **`ParsedDocsConfiguration.ts`**: Added `translationNavigationOverlays` field and full type hierarchy (`TranslationNavigationOverlay`, `ProductOverlay`, `VersionOverlay`, `TabOverlay`, `AnnouncementOverlay`, `NavigationItemOverlay`, `VariantOverlay`)
- **`parseDocsConfiguration.ts`**: Added `loadTranslationNavigationOverlays()` function that loads and parses overlay YAML files alongside the existing `loadTranslationPages()`
- **`applyTranslatedNavigationOverlays.ts`**: New nav tree walker that applies overlay overrides to product titles, version names, tab display-names, section headings, page titles, and announcements
- **`publishDocs.ts`**: Wired navigation overlays into per-locale `DocsDefinition` construction
- **`runAppPreviewServer.ts` / `runPreviewServer.ts` / `previewDocs.ts`**: Wired navigation overlays into local preview translated definitions
- **`DocsDefinitionResolver.ts`**: Exposed `getTranslationNavigationOverlays()` accessor

## Testing

- [x] Unit tests added (`applyTranslatedNavigationOverlays.test.ts` — 9 tests covering products, versions, tabs, sections/pages, announcements, and edge cases)
- [x] All tests pass
- [x] Lint and format checks pass

Link to Devin session: https://app.devin.ai/sessions/26c0bb49462141559dc4e5d28995b904
Requested by: @dsinghvi

### CODE REVIEW

**Comment by dsinghvi** [comment]
Does the translations overlay require that we have keys like `path: ./nav.yml` or `slug` ? b/c ideally those would be excluded and we would limit down to just the display names

**Comment by dsinghvi** [comment]
@claude review

**Comment by fern-support** [comment]
I've pushed a fix (d2209a8) that addresses the review feedback:

### 1. Path Traversal Security Fix
Fixed in `parseDocsConfiguration.ts`: The `productObj.path` is now validated using `path.resolve()` and checked to ensure it doesn't escape the translations directory before being used to load nav files.

### 2. Positional Matching Bug Fix
Fixed in `applyTranslatedNavigationOverlays.ts`: The `matchSectionOverlay` and `matchPageOverlay` functions now only use overlays **without** slugs for positional fallback. This prevents incorrectly applying a slug-targeted overlay to the wrong sibling when there's no slug match.

**Before (buggy)**: With `children=[A(slug=api), B(slug=foo), C(slug=guide)]` and `overlays=[{slug:api}, {slug:guide}]`, B would incorrectly get the `guide` overlay via positional fallback.

**After (fixed)**: B gets no overlay (correct), A matches `api` by slug, C matches `guide` by slug. Only overlays without slugs can be used for positional matching.

All 44 tests pass.

File: packages/cli/configuration-loader/src/docs-yml/parseDocsConfiguration.ts [hunk 1]
```diff
@@ -2028,3 +2040,275 @@ async function loadTranslationPages({
...
+            if (isPlainObject(productObj.announcement)) {
+                const message = extractString(productObj.announcement, "message");
+                if (message != null) {
+                    productOverlay.announcement = { message };
+                }
+            }
+
+            // If this product references a nav file, load its tab/navigation overrides
+            if (typeof productObj.path === "string") {
+                const navFilePath = path.join(langFernDir, productObj.path) as AbsoluteFilePath;
```
[the remark this answers is not in the document]
Comment (fern-support) [reply packages/cli/configuration-loader/src/docs-yml/parseDocsConfiguration.ts]: Fixed in d2209a8. The path is now validated with `path.resolve()` and checked to ensure it starts with `langFernDir` before being used. Invalid paths that escape the translations directory are logged as warnings and skipped.

File: packages/cli/docs-resolver/src/applyTranslatedNavigationOverlays.ts [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+    for (const o of overlays) {
+        if (o.slug != null && o.slug === sectionSlug) {
+            return o;
+        }
+    }
+
+    if (positionIndex < overlays.length) {
+        return overlays[positionIndex];
+    }
+    return undefined;
```
[the remark this answers is not in the document]
Comment (fern-support) [reply packages/cli/docs-resolver/src/applyTranslatedNavigationOverlays.ts:419]: Fixed in d2209a8. The positional fallback now only considers overlays that don't have a `slug` defined. This ensures slug-targeted overlays are only matched by slug, preventing the incorrect double-application scenario you described.

**Comment by fern-support** [comment]
## Additional Fixes (939b848)

Addressed all remaining review feedback:

### 1. Multi-product tab/navigation collision fix
**Files**: `ParsedDocsConfiguration.ts`, `parseDocsConfiguration.ts`, `applyTranslatedNavigationOverlays.ts`

Added `tabs` and `navigation` fields to `ProductOverlay` so each product stores its own overlays instead of merging them globally. When processing products, a scoped overlay is now created with the product-specific tabs/navigation.

### 2. sidebarGroup handling fix
**File**: `applyTranslatedNavigationOverlays.ts`

Added a `sidebarGroup` branch in `applyChildOverlays` that delegates to `applySidebarChildOverlays`. This ensures top-level pages wrapped in sidebarGroup nodes (by `DocsDefinitionResolver`) get translated.

### 3. Tab variant overlays applied
**File**: `applyTranslatedNavigationOverlays.ts`

Added `applyVariantOverlays` function and integrated it into `applyTabOverlayToNode`. Tab variants now have their `title` and `subtitle` translated via slug or positional matching.

### 4. Dead code removed
**File**: `applyTranslatedNavigationOverlays.ts`

Removed the `displayName` fallback branch in `findTabNavOverlay` that compared translated labels against source-language titles (which would never match for real translations). Also removed the unused `tabTitle` parameter.

All 44 tests pass.
